nservicebus/handlers/index.md

```diff
@@ -45,18 +45,18 @@ partial: behaviorcaveat
 
 ## Invocation of multiple matching handlers
 
-When an endpoint hosts multiple message handlers which all "match" a single incoming message and one of the handlers fails then the incoming message gets retried. When the incoming message is retried all matching handlers get invoked again. Invoked handlers include the handlers that might have already been successfully invoked in previous attempts.
+It is important to keep in mind that, by default, a single incoming message will be processed as a single unit of work, regardless of the number of handlers which are hosted in that endpoint. If one of the handlers fails, the incoming message will be retried according to the [recoverability policy of the endpoint](/nservicebus/recoverability). When the incoming message is retried, _all_ matching handlers get invoked again, including the handlers that had successfully processed the message in previous attempts.
```

Comment:
I don't understand. The word "should" here would undermine the "by default" earlier in the sentence.

Reply:
@udidahan it could be that its not processed as an atomic unit-of-work. To me 'will' feels incorrect.

So maybe it should be the following:
```suggestion
It is important to keep in mind that a single incoming message should be processed as a single unit of work, regardless of the number of handlers which are hosted in that endpoint. If one of the handlers fails, the incoming message will be retried according to the [recoverability policy of the endpoint](/nservicebus/recoverability). When the incoming message is retried, _all_ matching handlers get invoked again, including the handlers that had successfully processed the message in previous attempts.
```